test/export/mcap_test.py

```diff
@@ -0,0 +1,221 @@
+# BSD 3-Clause License
+#
+# Copyright (c) 2023, Woven by Toyota
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# 3. Neither the name of the copyright holder nor the names of its
+#    contributors may be used to endorse or promote products derived from
+#    this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+import os
+import unittest
+
+import shutil
+from unittest.mock import MagicMock
+from typing import List
+from tempfile import mkdtemp
+
+from maliput_sim.core.sim import SimulationState
+from maliput_sim.simulation import Simulation
+from maliput_sim.export.mcap import MCAP
+
+from foxglove_schemas_protobuf.SceneUpdate_pb2 import SceneUpdate
+from foxglove_schemas_protobuf.Pose_pb2 import Pose
+from foxglove_schemas_protobuf.Quaternion_pb2 import Quaternion
+from foxglove_schemas_protobuf.Vector3_pb2 import Vector3
+from maliput.plugin import create_road_network
+from mcap_protobuf.decoder import DecoderFactory
+from mcap.reader import make_reader
+
+
+def GetRoadNetwork():
+    """
+    Returns an arbitrary road network.
+    """
+    # TODO(#12): Rely on a road network loader once that is implemented.
+    # Create road network model
+    env = os.getenv("MALIPUT_MALIDRIVE_RESOURCE_ROOT").split(":")
+    env = env[0]
+    rn_configuration = {
+        "opendrive_file": env + "/resources/odr/TShapeRoad.xodr" if env is not None else "",
+        "road_geometry_id": "TShapeRoadRoadNetwork"
+    }
+    maliput_road_network = create_road_network(
+        "maliput_malidrive", rn_configuration)
+    return maliput_road_network
```

Comment:
My first approach was to mock the road network and to [patch](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch) the call to `maliput_sim.utils.obj.get_obj_description(road_network)`  (which happens within mcap exporter) for returning a known obj description instead of having to load up the road network.

For a reason that I don't know I couldn't make it work, the patch didn't work out, so I ended up loading up a road network.

Reply:
It's fine. We can revisit this in the future if required.